cuml/kmeans_demo.ipynb

```diff
@@ -0,0 +1,233 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# KMeans\n",
+    "\n",
+    "KMeans is a basic but powerful clustering method which is optimized via Expectation Maximization. It randomnly selects K data points in X, and computes which samples are close to these points. For every cluster of points, a mean is computed (hence the name), and this becomes the new centroid.\n",
+    "\n",
+    "cuML’s KMeans supports the fast KMeans++ intialization method. This method is more stable than randomnly selecting K points.\n",
+    "\n",
+    "<br>\n",
+    "\n",
+    "|  Parameters: |  Description:    |\n",
+    "| ------------ | ---------------- |\n",
+    "| `n_clusters:int (default = 8)` | The number of centroids or clusters you want. |\n",
+    "| `max_iter:int (default = 300)` | The more iterations of EM, the more accurate, but slower. |\n",
+    "| `tol:float (default = 1e-4)` | Stopping criterion when centroid means do not change much. |\n",
+    "| `verbose:boolean (default = 0)` | If True, prints diagnositc information. |\n",
+    "| `random_state:int (default = 1)` | If you want results to be the same when you restart Python, select a state. |\n",
+    "| `precompute_distances:boolean (default = ‘auto’)` | Not supported yet. |\n",
+    "| `init: (default = 'scalable-k-means++')` <br> -`'scalable-k-means++' or 'k-meansǀǀ'`  <br> -`'random' or an ndarray` |  <br> - Uses fast and stable scalable kmeans++ intialization. <br>  - Choose 'n_cluster' observations (rows) at random from data for the initial centroids. <br>   If an ndarray is passed, it should be of shape (n_clusters, n_features) and gives the initial centers. |\n",
+    "| `n_init:int (default = 1)` | Number of times intialization is run. More is slower, but can be better. |\n",
+    "| `algorithm:“auto”` | Currently uses full EM, but will support others later. |\n",
+    "\n",
+    "<br>\n",
+    "\n",
+    "|  Methods:         |  Description:      |\n",
+    "| ------------------| -----------------  |\n",
+    "|`fit(self, X)` | Compute k-means clustering with X. |\n",
+    "|`fit_predict(self, X)` | Compute cluster centers and predict cluster index for each sample. |\n",
+    "|`fit_transform(self, input_gdf)` | Compute clustering and transform input_gdf to cluster-distance space. |\n",
+    "|`get_params(self[, deep])` | Sklearn style return parameter state |\n",
+    "|`predict(self, X)` | Predict the closest cluster each sample in X belongs to. |\n",
+    "|`set_params(self, **params)` | Sklearn style set parameter state to dictionary of params. |\n",
+    "|`transform(self, X)` | Transform X to a cluster-distance space. |\n",
+    "\n",
```

Comment:
Lets not duplicate the description of parameter and methods. Just adding a pointer to rapidsai API documentation should suffice, that way if the documentation is updated then it will be in one place.

Reply:
Okay, this makes the most sense to me. Will change.